### PR TITLE
Promote resource limits priority function to beta

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What this PR does / why we need it**:
We should promote resource limits priority function to beta as this feature is useful in scenarios where the nodes that satisfies limits of pods are given higher score.

/cc @aveshagarwal @bsalamat 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Promote resource limits priority function to beta
```

### CODE REVIEW

**Review by bsalamat — COMMENTED** [review]
I am fine with the promotion, I just wonder if you have tested this feature in real clusters enough.

File: test/e2e/scheduling/priorities.go
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: I think you need to revert the node resources back to avoid surprises for future test cases.

File: test/e2e/scheduling/priorities.go
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: Any particular reason for deleting this?

Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Thanks for catching, you are right, I was thinking I was changing the test I am introducing as part of this PR. I was not intending to touch this.

File: test/e2e/scheduling/priorities.go
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: This was part of testing, I was doing, I forgot to revert it. Thanks.

File: test/e2e/scheduling/priorities.go
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Yes, I missed it. I have updated the PR now.

File: test/e2e/scheduling/priorities.go
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Sure, I have added it.

File: test/e2e/scheduling/priorities.go
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: nit: s/value/memory/

Also, please add a comment for the function and perhaps make it private (change to lower case).

File: test/e2e/scheduling/priorities.go
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: nit: This if statement is not needed. At this point it returns 'err' regardless.

File: test/e2e/scheduling/priorities.go
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: Shouldn't you read the original amount of memory at the beginning of the function, before you update it?

**Comment by yastij** [comment]
cc @kubernetes/api-approvers

File: test/e2e/scheduling/priorities.go
Comment (k82cn) [reply test/e2e/scheduling/priorities.go]: > so actually this could have failed after the PR merged :(.

I mean set `nodeName` to middle of `nodeList.Items`:) 
As I think you're setting a smaller memory (10^4) than pod's limits (3 * 10^), it's better to use a const, e.g. node's memory is twice of big pod.

File: test/e2e/scheduling/priorities.go
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: Looks like this var is used only in one of the tests. Please move it inside the test func.

**Review by bsalamat — COMMENTED** [review]
just a minor comment. Otherwise, LGTM.

**Review by bsalamat — APPROVED** [review]
/lgtm

Thanks, @ravisantoshgudimetla!

**Comment by AishSundar** [comment]
@ravisantoshgudimetla I see the newly added Scheduler Priorities test added fail in the test run - https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gke-gci-new-gci-master-upgrade-cluster-new

Can you please take a look?

@jberkus @mortent as FYI. We need to open a test issue to track the test failure if it repros in the next run too.

**Comment by ravisantoshgudimetla** [comment]
@AishSundar Sure, I will look into it. FWIW, the test is also failing on master(https://k8s-testgrid.appspot.com/sig-scheduling#gce-serial).

**Comment by AishSundar** [comment]
/milestone v1.13

Ok lets open a new test issue for this. thanks @ravisantoshgudimetla

**Comment by bsalamat** [comment]
@ravisantoshgudimetla Looks like the e2e test for this PR is flaky: https://k8s-testgrid.appspot.com/sig-release-master-blocking#gce-cos-master-serial

Could you please take a look?

**Pull request merged**
